sig-and-wg/sig-core/proposals/knative_function_integration.md

```diff
@@ -0,0 +1,361 @@
+- As a developer, I want to view the status of my function.
+
+## Proposed Solution
+
+### Overview
+We are proposing introducing the following changes:
+
+1. A new `Function` CRD and `function-controller` component.
+2. Bundle [Tekton Pipelines](https://github.com/tektoncd/pipeline/) and [Knative Serving](https://github.com/knative/serving) as Kyma modules.
+3. Bundle a docker registry inside Kyma for hosting function images.
+4. Adding GraphQL layer support.
+5. Updating the UI with a new Microfrontend Functions section that leverages all new functionality.
+
+### Function CRD
+
+#### CRD Definition
+
+The new `Function` CRD gives the user a way to define how to turn his source code into a running container inside Kyma. That source code is conventionally a single function running inside a specific execution environment (aka runtime). This is an example of the minimum set of fields you need to successfully define your function.
+
+```YAML
+apiVersion: functions.serverless.kyma-project.io/v1alpha1
+kind: Function
+metadata:
+  name: sample
+  labels:
+    foo: bar
+spec:
+  function: |
+    module.exports = {
+        main: function(event, context) {
+          return 'Hello World'
+        }
+      }
+  functionContentType: "plaintext"
+  size: "L"
+  runtime: "nodejs8"
+```
+
+[Appendix A](#appendix-a-function-custom-resource-definition) has the full definition of the function CRD.
+
+#### Function Controller
+
+The function controller reconciles `Function` objects by using `Tekton` and `Knative Serving` to do the following:
+
+1. Create a `TaskRun` object for the proper `Task` based on the provided runtime in the `Function Object`.
+2. Create a `Knative Service` object that refers to the newly published image created in the previous step.
+
+The following diagram gives a brief overview of the sequence of events. The
+only thing the user does is create a function object.
+
+![function controller](assets/knative_function_controller.svg) .
+
+### Tekton Pipelines and Knative Serving
+
+`Tekton Pipelines` and `Knative Serving` will be bundled with Kyma as Kyma modules. They will be installed by default once the knative-functions provider completely replaces Kubeless.
+
+### Docker Registry module
+
+[Docker Registry](https://hub.docker.com/_/registry) will be bundled as a Kyma module that is used to host and serve the docker images generated by building functions. It will be configurable to use the Minio in Kyma for storage. It'll also be configurable so that Kyma
+users can skip installing it and provide an external registry instead.
+
+### GraphQL layer support
+
+For the user to have a fully functional and integrated function from the UI, CRUD operations for the following four
+objects needs to be supported in the GraphQL API layer.
+
+* Function
+* ServiceBinding
+* ServiceBindingUsage
+* API  
+
+
+#### Function GraphQL Objects Spec
```

Comment:
I am not sure how this would be done. Since we would additionally need to pass `api`, `events` and `servicebinding` info

Reply:
so my understanding from @k15r is that these are the "missing" objects, so api, service binding, events etc all have existing objects. it will be several objects.